### PR TITLE
Video UI Changes

### PR DESCRIPTION
**Pull request by christianmemije**
## Summary

A few UI changes to the video player. This is still work in progress and would appreciate feedback on design choices.

* Removed the buttons from the middle since they were occupying too much space. Instead moved them to the bottom.
* Made the seek bar it's own line so it would allow more room for the components. 
* Grouped the time together to keep things simpler. 
* Replaced the time remaining with duration component instead.
* Added text to the replay and forward buttons.
* Made the volume component vertical to save space.
* Changed the colors just to keep it simple.
* Made the buttons slightly smaller on smaller devices so they don't overlap.

## TODO

- [x] Test on IE9 and smaller devices. Styling a bit off on IE 9, but still fully functional. 

### Before
before

### Now
image

### CODE REVIEW

File: kolibri/plugins/video_mp4_render/assets/src/vue/videojs-replay-forward-btns.js [hunk 1]
```diff
@@ -21,21 +21,7 @@ class ForwardButton extends videojsButton {
...
-    this.toggleClass('videopaused');
-    if (player.paused()) {
-      player.play();
-    } else {
-      player.pause();
-    }
-  }
-}
-
+ReplayButton.prototype.controlText_ = 'Replay';
```
Comment (rtibbles) [inline kolibri/plugins/video_mp4_render/assets/src/vue/videojs-replay-forward-btns.js]: I assume this is user facing text? We need to think about how to make this internationalizable..

Comment (rtibbles) [reply kolibri/plugins/video_mp4_render/assets/src/vue/videojs-replay-forward-btns.js]: Made a PR to address this: https://github.com/christianmemije/kolibri/pull/8

We will also have to look at how best to bundle videojs' pretranslated strings etc.

**Comment by rtibbles** [comment]
Will merge once my PR is merged for the string access.

File: kolibri/plugins/video_mp4_render/assets/src/vue/videojs-replay-forward-btns.js (hunk 1 above)
Comment (christianmemije) [reply kolibri/plugins/video_mp4_render/assets/src/vue/videojs-replay-forward-btns.js]: How VideoJS does it: https://github.com/videojs/video.js/blob/master/docs/guides/languages.md

**Comment by christianmemije** [comment]
Need to fix some styling on IE9 and FIrefox.

**Comment by jtamiace** [comment]
I like the change to the seek bar and time! however, i feel like the play button on video players has a very standardized placement on its UI (unless there are platforms/players that put play/pause in the center). 

what about this alignment?
slack for ios upload

**Comment by christianmemije** [comment]
@jtamiace I agree that it is standard to put the play button on the left on desktop sites. However, I noticed a lot of mobile apps put the play button on the center surrounded by the rewind and forward buttons, which could work well with our replay and forward buttons.

**Comment by christianmemije** [comment]
@radinamatic 
image

**Comment by christianmemije** [comment]
Ready to be merged. Not _too_ bad on IE 9:
image

**Comment by radinamatic** [comment]
What I noticed so far:

1. Once video element has focus, I can start it with both `Enter` and `Spacebar`, but once started it does not stop on pressing `Enter` again, only the `Spacebar` can pause/stop it. 

2. While testing this I keep getting the `Network error` on both FF and Chrome, but in FF it just displays in the Console, while in Chrome I had to reload in order to try playing again:

kolibri - google chrome_125

3. Maybe we should find alternative for  `'KeyboardEvent.keyIdentifier'` because of the warning   `'KeyboardEvent.keyIdentifier' is deprecated and will be removed in M54, around October 2016.`?

4. CSS focus style for elements inside the player control bar is way to subdued to actually be useful - one has to be Hawkeye Bumppo to notice it... :unamused: . I could try to change it myself, just not sure which CSS file should I mess with...?

**Comment by christianmemije** [comment]
1. Yes that is correct behavior. When you press enter, a click event on the big play button is fired, which disappears once the video starts playing, so there is nothing in focus to handle the space.
2. Can you reproduce these network issues on a non-dev-server build?
3. That is from the [videojs code](https://github.com/videojs/video.js/blob/b4265a8468fa1d685ea977a515243d6443603e3d/src/js/utils/events.js#L105). 
4. Don't get the reference, but I will try to make the focus more noticeable.

**Comment by radinamatic** [comment]
I'll try to reproduce on non-dev environment, this is the server output in dev:

```
[03/Oct/2016 14:58:34] "POST /api/contentsessionlog/ HTTP/1.1" 201 250
[03/Oct/2016 14:59:19] "GET /content/storage/6/3/635e925e41553f4eb207523f48f8c251.mp4 HTTP/1.1" 200 188416
Traceback (most recent call last):
  File "/usr/lib/python3.4/wsgiref/handlers.py", line 138, in run
    self.finish_response()
  File "/usr/lib/python3.4/wsgiref/handlers.py", line 180, in finish_response
    self.write(data)
  File "/usr/lib/python3.4/wsgiref/handlers.py", line 279, in write
    self._write(data)
  File "/usr/lib/python3.4/wsgiref/handlers.py", line 453, in _write
    self.stdout.write(data)
  File "/usr/lib/python3.4/socket.py", line 394, in write
    return self._sock.send(b)
ConnectionResetError: [Errno 104] Connection reset by peer
[03/Oct/2016 14:59:19] "GET /content/storage/6/3/635e925e41553f4eb207523f48f8c251.mp4 HTTP/1.1" 500 59
----------------------------------------
Exception happened during processing of request from ('127.0.0.1', 41042)
Traceback (most recent call last):
  File "/usr/lib/python3.4/wsgiref/handlers.py", line 138, in run
    self.finish_response()
  File "/usr/lib/python3.4/wsgiref/handlers.py", line 180, in finish_response
    self.write(data)
  File "/usr/lib/python3.4/wsgiref/handlers.py", line 279, in write
    self._write(data)
  File "/usr/lib/python3.4/wsgiref/handlers.py", line 453, in _write
    self.stdout.write(data)
  File "/usr/lib/python3.4/socket.py", line 394, in write
    return self._sock.send(b)
ConnectionResetError: [Errno 104] Connection reset by peer

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.4/wsgiref/handlers.py", line 141, in run
    self.handle_error()
  File "/home/didi/.virtualenvs/kolibri/lib/python3.4/site-packages/django/core/servers/basehttp.py", line 92, in handle_error
    super(ServerHandler, self).handle_error()
  File "/usr/lib/python3.4/wsgiref/handlers.py", line 368, in handle_error
    self.finish_response()
  File "/usr/lib/python3.4/wsgiref/handlers.py", line 180, in finish_response
    self.write(data)
  File "/usr/lib/python3.4/wsgiref/handlers.py", line 274, in write
    self.send_headers()
  File "/usr/lib/python3.4/wsgiref/handlers.py", line 331, in send_headers
    if not self.origin_server or self.client_is_modern():
  File "/usr/lib/python3.4/wsgiref/handlers.py", line 344, in client_is_modern
    return self.environ['SERVER_PROTOCOL'].upper() != 'HTTP/0.9'
TypeError: 'NoneType' object is not subscriptable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.4/socketserver.py", line 617, in process_request_thread
    self.finish_request(request, client_address)
  File "/usr/lib/python3.4/socketserver.py", line 344, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/home/didi/.virtualenvs/kolibri/lib/python3.4/site-packages/django/core/servers/basehttp.py", line 99, in __init__
    super(WSGIRequestHandler, self).__init__(*args, **kwargs)
  File "/usr/lib/python3.4/socketserver.py", line 673, in __init__
    self.handle()
  File "/home/didi/.virtualenvs/kolibri/lib/python3.4/site-packages/django/core/servers/basehttp.py", line 179, in handle
    handler.run(self.server.get_app())
  File "/usr/lib/python3.4/wsgiref/handlers.py", line 144, in run
    self.close()
  File "/usr/lib/python3.4/wsgiref/simple_server.py", line 35, in close
    self.status.split(' ',1)[0], self.bytes_sent
AttributeError: 'NoneType' object has no attribute 'split'
----------------------------------------
[03/Oct/2016 14:59:22] "PATCH /api/contentsessionlog/99/ HTTP/1.1" 200 293
[03/Oct/2016 14:59:45] "GET /learn/ HTTP/1.1" 200 15703
[03/Oct/2016 14:59:49] "GET /api/session/current/ HTTP/1.1" 200 116
[03/Oct/2016 14:59:49] "GET /api/content/ HTTP/1.1" 200 618
[03/Oct/2016 14:59:49] "GET /api/content/8b4d3e6d3d4842ba8ea658335b5dd252/contentnode/0ed4651073a9449fb29eb3875c0b8ef9/ HTTP/1.1" 200 1645
[03/Oct/2016 14:59:49] "GET /static/images/logo.ico HTTP/1.1" 200 5430
[03/Oct/2016 14:59:52] "GET /content/storage/6/3/635e925e41553f4eb207523f48f8c251.mp4 HTTP/1.1" 200 1901768
[03/Oct/2016 14:59:52] "POST /api/contentsessionlog/ HTTP/1.1" 201 251
[03/Oct/2016 15:00:26] "GET /content/storage/6/3/635e925e41553f4eb207523f48f8c251.mp4 HTTP/1.1" 200 630784
Traceback (most recent call last):
  File "/usr/lib/python3.4/wsgiref/handlers.py", line 138, in run
    self.finish_response()
  File "/usr/lib/python3.4/wsgiref/handlers.py", line 180, in finish_response
    self.write(data)
  File "/usr/lib/python3.4/wsgiref/handlers.py", line 279, in write
    self._write(data)
  File "/usr/lib/python3.4/wsgiref/handlers.py", line 453, in _write
    self.stdout.write(data)
  File "/usr/lib/python3.4/socket.py", line 394, in write
    return self._sock.send(b)
ConnectionResetError: [Errno 104] Connection reset by peer
[03/Oct/2016 15:00:26] "GET /content/storage/6/3/635e925e41553f4eb207523f48f8c251.mp4 HTTP/1.1" 500 59
----------------------------------------
Exception happened during processing of request from ('127.0.0.1', 41066)
Traceback (most recent call last):
  File "/usr/lib/python3.4/wsgiref/handlers.py", line 138, in run
    self.finish_response()
  File "/usr/lib/python3.4/wsgiref/handlers.py", line 180, in finish_response
    self.write(data)
  File "/usr/lib/python3.4/wsgiref/handlers.py", line 279, in write
    self._write(data)
  File "/usr/lib/python3.4/wsgiref/handlers.py", line 453, in _write
    self.stdout.write(data)
  File "/usr/lib/python3.4/socket.py", line 394, in write
    return self._sock.send(b)
ConnectionResetError: [Errno 104] Connection reset by peer

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.4/wsgiref/handlers.py", line 141, in run
    self.handle_error()
  File "/home/didi/.virtualenvs/kolibri/lib/python3.4/site-packages/django/core/servers/basehttp.py", line 92, in handle_error
    super(ServerHandler, self).handle_error()
  File "/usr/lib/python3.4/wsgiref/handlers.py", line 368, in handle_error
    self.finish_response()
  File "/usr/lib/python3.4/wsgiref/handlers.py", line 180, in finish_response
    self.write(data)
  File "/usr/lib/python3.4/wsgiref/handlers.py", line 274, in write
    self.send_headers()
  File "/usr/lib/python3.4/wsgiref/handlers.py", line 331, in send_headers
    if not self.origin_server or self.client_is_modern():
  File "/usr/lib/python3.4/wsgiref/handlers.py", line 344, in client_is_modern
    return self.environ['SERVER_PROTOCOL'].upper() != 'HTTP/0.9'
TypeError: 'NoneType' object is not subscriptable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.4/socketserver.py", line 617, in process_request_thread
    self.finish_request(request, client_address)
  File "/usr/lib/python3.4/socketserver.py", line 344, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/home/didi/.virtualenvs/kolibri/lib/python3.4/site-packages/django/core/servers/basehttp.py", line 99, in __init__
    super(WSGIRequestHandler, self).__init__(*args, **kwargs)
  File "/usr/lib/python3.4/socketserver.py", line 673, in __init__
    self.handle()
  File "/home/didi/.virtualenvs/kolibri/lib/python3.4/site-packages/django/core/servers/basehttp.py", line 179, in handle
    handler.run(self.server.get_app())
  File "/usr/lib/python3.4/wsgiref/handlers.py", line 144, in run
    self.close()
  File "/usr/lib/python3.4/wsgiref/simple_server.py", line 35, in close
    self.status.split(' ',1)[0], self.bytes_sent
AttributeError: 'NoneType' object has no attribute 'split'
----------------------------------------
[03/Oct/2016 15:00:28] "PATCH /api/contentsessionlog/100/ HTTP/1.1" 200 295
[03/Oct/2016 15:01:37] "GET /learn/ HTTP/1.1" 200 15686
```

**Comment by radinamatic** [comment]
> Yes that is correct behavior. When you press enter, a click event on the big play button is fired, which disappears once the video starts playing, so there is nothing in focus to handle the space.

Might be correct, but it's misleading for the user who expects that if the video started by pressing Enter, it will stop when the same button is pressed again... Would it be possible to set focus to the `Play` button inside the player control bar, immediately after the video starts playing?

Furthermore, when tab-navigating buttons inside the player control bar, it is not possible to use Spacebar to start & stop, one must use Enter. We should strive to make the behavior consistent, so either it's possible to use **only** Enter or **only** Spacebar or **both** of them consistently to start/pause/stop the playback.

> That is from the videojs code.

So we are confident that they will release an update in time? :stuck_out_tongue: 

> Don't get the reference, but I will try to make the focus more noticeable.

Apologies, [too obscure](https://en.wikipedia.org/wiki/Natty_Bumppo#Aliases)... :blush:

**Comment by christianmemije** [comment]
> Might be correct, but it's misleading for the user who expects that if the video started by pressing Enter, it will stop when the same button is pressed again... Would it be possible to set focus to the Play button inside the player control bar, immediately after the video starts playing?



> Furthermore, when tab-navigating buttons inside the player control bar, it is not possible to use Spacebar to start & stop, one must use Enter. We should strive to make the behavior consistent, so either it's possible to use only Enter or only Spacebar or both of them consistently to start/pause/stop the playback.

Good catch! That is due to the spacebar hotkey... Two play toggle events are fired which causes it to appear to do nothing. Will fix this.

> So we are confident that they will release an update in time? :stuck_out_tongue:

They are checking that it does not exist, for the sake of older browsers. It isn't code that would break so we shouldn't worry about the warning.

**Comment by christianmemije** [comment]
Will remove hotkeys for now until I plan out a correct implementation of hotkeys.

**Comment by christianmemije** [comment]
Took into the account some of the feedback that @jtamiace and @radinamatic and kept the video player more standard. Now looks like this:
image

**Comment by christianmemije** [comment]
@indirectlylit Ready for review.

**Comment by radinamatic** [comment]
@christianmemije Nice :+1: 

Too late for me to test locally, but I'll review first thing tomorrow!

**Comment by christianmemije** [comment]
@radinamatic Thanks! By the way, were you ever able to reproduce the errors on a non-dev server?

**Comment by indirectlylit** [comment]
I like these changes!

it's an edge case that we can address in a future issue/PR, but videos with weird aspect ratios introduce some breakage:

image

**Comment by indirectlylit** [comment]
also, please sync this with `develop`

**Comment by christianmemije** [comment]
@indirectlylit I did notice that too, it has to due to the containers that wrap the video player which I can look at. Also, just synced with dev.

**Comment by radinamatic** [comment]
Ok, several comments:

1. Looks to me as if upon first page load in Firefox the video player container itself can get focus, while in Chrome it cannot (in it navigating through elements with TAB key after the `Search` button focus goes directly to the central `Play` button). This creates confusion for the FF user as  one additional TAB-step is necessary to start the playback. Is it possible to avoid this in FF?

2. When user starts playing the video with that big central button, focus seems to be lost somewhere in the limbo and video cannot be stopped neither with Enter nor Spacebar - one additional TAB step is necessary to stop it. This happens on both FF & Chrome. Would it be possible to shift focus to the `Play` button **in the control bar** once video is started with the big central button? That way user can start/stop the playback without tabbing in between?

3. Not sure if that's something you did in this PR, but the `Back to previous topic` folder button seems misplaced: selection_147

4. You were right about the previous error - no sign of it inside production server... :+1:

**Comment by christianmemije** [comment]
@radinamatic Thanks for the feedback.

I see that, do you know on exactly what element that extra focus is on? Or any helpful FF extensions for testing tabbing?

I will do that.

That's something unrelated to my code. Should not exist in the latest dev branch.

**Comment by radinamatic** [comment]
> Or any helpful FF extensions for testing tabbing?

You could try [Firebug](http://getfirebug.com/)/[FireFocus](http://www.incaseofstairs.com/firefocus/) combo.

**Comment by christianmemije** [comment]
@radinamatic Addressed both issues in those 2 commits.

**Comment by christianmemije** [comment]
@indirectlylit Now:

image

Should be ready to merge. :tada:

**Comment by indirectlylit** [comment]
Looks like you've got a few linting errors: https://travis-ci.org/learningequality/kolibri/jobs/169011147

**Comment by radinamatic** [comment]
@christianmemije Awesome job!!! :100: 
Keyboard navigation in FF flies and purrs... :smiley: 

Buttons to skip back & forward and clicking on progress bar do not work in Chrome atm: they all restart the playback. Was that an issue before too?

**Comment by christianmemije** [comment]
@indirectlylit Fixed the lint error. @radinamatic that is an issue that occurs on the devserver, nothing new. :/

**Comment by christianmemije** [comment]
@indirectlylit Looks like there is still an error but doesn't look related to this...

**Pull request merged**